File: src/main/java/com/pack/colorchat/Color.java
package com.pack.colorchat;

import java.util.Arrays;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

public enum Color {
    GREEN("green", "Green"),
    RED("red", "Red"),
    YELLOW("yellow", "Yellow"),
    BLUE("blue", "Blue"),
    BLACK("black", "Black");

    private static final Map<String, Color> VALUES;

    static {
        Map<String, Color> values = new HashMap<>();
        Arrays.stream(values()).forEach(color -> values.put(color.id, color));
        VALUES = Collections.unmodifiableMap(values);
    }

    private final String id;
    private final String text;

    Color(String id, String text) {
        this.id = id;
        this.text = text;
    }

    public static Color of(String color) {
        return VALUES.getOrDefault(color, BLACK);
    }

    public String getId() {
        return id;
    }

    public String getText() {
        return text;
    }
}

File: src/main/kotlin/com/pack/colorchat/model/Color.kt
package com.pack.colorchat.model

import java.util.*

enum class Color(val id: String, val text: String) {
    GREEN("green", "Green"),
    RED("red", "Red"),
    YELLOW("yellow", "Yellow"),
    BLUE("blue", "Blue"),
    BLACK("black", "Black");


    companion object {

        private val VALUES: Map<String, Color>

        init {
            val colors = HashMap<String, Color>()
            Arrays.stream(values()).forEach { colors.put(it.id, it) }
            VALUES = Collections.unmodifiableMap(colors)
        }

        fun of(color: String): Color = VALUES.getOrDefault(color, BLACK)
    }
}
